#!/usr/bin/env python
PACKAGE = "gripper"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("depth_P_gain", double_t, 0, "Depth P gain", 3, 0, 10)
gen.add("depth_I_gain", double_t, 0, "Depth I gain", 0.4, 0, 10)
gen.add("depth_D_gain", double_t, 0, "Depth D gain", 2.5, 0, 10)
gen.add("vertical_thrust_max", double_t, 0, "Vertical thrust max", 0.4, 0, 1)

gen.add("thrust_P_gain", double_t, 0, "thrust P gain", 0.4, 0, 10)
gen.add("thrust_I_gain", double_t, 0, "thrust I gain", 0.1, 0, 10)
gen.add("thrust_D_gain", double_t, 0, "thrust D gain", 0, 0, 10)
gen.add("thrust_max", double_t, 0, "thrust max", 0.3, 0, 1)

gen.add("lateral_thrust_P_gain", double_t, 0, "lateral thrust P gain", 0.2, 0,10)
gen.add("lateral_thrust_I_gain", double_t, 0, "lateral thrust I gain", 0.2, 0, 10)
gen.add("lateral_thrust_D_gain", double_t, 0, "lateral thrust D gain", 0.2, 0, 10)
gen.add("lateral_thrust_max", double_t, 0, "lateral thrust max", 0.4, 0, 1)

gen.add("roll_P_gain", double_t, 0, "roll P gain", 0.2, 0, 10)
gen.add("roll_I_gain", double_t, 0, "roll I gain", 0.2, 0, 10)
gen.add("roll_D_gain", double_t, 0, "roll D gain", 0.2, 0, 10)
gen.add("roll_max", double_t, 0, "roll max", 0.4, 0, 1)

gen.add("pitch_P_gain", double_t, 0, "pitch P gain", 0.7, 0, 10)
gen.add("pitch_I_gain", double_t, 0, "pitch I gain", 0.7, 0, 10)
gen.add("pitch_D_gain", double_t, 0, "pitch D gain", 0.2, 0, 10)
gen.add("pitch_max", double_t, 0, "pitch max", 0.7, 0, 1)

gen.add("yaw_P_gain", double_t, 0, "yaw P gain", 0.8, 0, 10)
gen.add("yaw_I_gain", double_t, 0, "yaw I gain", 0.2, 0, 10)
gen.add("yaw_D_gain", double_t, 0, "yaw D gain", 0, 0, 10)
gen.add("yaw_max", double_t, 0, "yaw max", 0.3, 0, 1)

gen.add("max_integrator", double_t, 0, "max integrator", 1, 0, 20)

exit(gen.generate(PACKAGE, "test_node", "PathFinder"))
